Dynatrace/dynatrace-operator
# WIP Ensure ActiveGate is Enabled for Extensions

## Description

see [K8S-10118](https://dt-rnd.atlassian.net/browse/K8S-10118)

ActiveGate is also created when Extensions are enabled in the DynaKube. No other AG capabilities are needed.

Dynakube v1beta2 is manually converted to v1beta3 in all cases when `dynakube.NeedsActiveGate()` or `NeedsActiveGateService()` is called.

## How can this be tested?

1) Unittests.
2) Apply dynakubes below. ActiveGate pod should be always created.
Check enabled capabilities on the AG pod using:
```
kubectl -n dynatrace get -o yaml pod/dynakube-activegate-0 -o jsonpath="{.spec.containers[0].env[0]}"
```

a)
dynakube:
```
spec:
  activeGate:
    capabilities:
      - routing
    replicas: 1
  extensions:
    prometheus:
      enabled: true
  oneAgent:
    cloudNativeFullStack: {}
```
capabilities:
```
{"name":"DT_CAPABILITIES","value":"MSGrouter,extension_controller"}
```

b)
dynakube:
```
spec:
  activeGate:
    replicas: 1
  extensions:
    prometheus:
      enabled: true
  oneAgent:
    cloudNativeFullStack: {}
```
capabilities:
```
{"name":"DT_CAPABILITIES","value":"extension_controller"}
```

c)
dynakube:
```
spec:
  extensions:
    prometheus:
      enabled: true
  oneAgent:
    cloudNativeFullStack: {}
```
capabilities:
```
{"name":"DT_CAPABILITIES","value":"extension_controller"}
```
